controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -53,15 +53,16 @@
      * Segment Table records are indexed and and it is O(constant) operation to get segment offset given segmentIndex.
      *
      * @param segmentId    segment id
+     * @param startingSegmentNumber starting segment number
      * @param segmentIndex segment table index
      * @param segmentTable segment table
      * @param historyIndex history index
      * @param historyTable history table
      * @return Segment object
      */
-    public static Segment getSegment(final long segmentId, final byte[] segmentIndex, final byte[] segmentTable,
-                                     final byte[] historyIndex, final byte[] historyTable) {
-        int segmentNumber = getSegmentNumber(segmentId);
+    public static Segment getSegment(final long segmentId, final int startingSegmentNumber, final byte[] segmentIndex,
+                                     final byte[] segmentTable, final byte[] historyIndex, final byte[] historyTable) {
+        int segmentNumber = getSegmentNumber(segmentId) - startingSegmentNumber;
```

Comment:
segment index only stores "offset in segment table". Segment number is implicit. But with starting segment number, we should ensure its semantics/apis stay clean and consistent. 
Currently that is only called from getSegment and there you are handling the segment number by substracting the starting number. But it would be cleaner if you use the actual segment number and change the api there to pass the starting segment number to SegmentIndex instead.

Reply:
in fact if we add starting segment number as first information in the index, then we dont even need to change the api. and we dont need to pass this information all the time.

Reply:
With the exception of `createSegmentTableAndIndex` (where the starting segment number is needed as input parameter to be stored in `segmentIndex`), the rest of `TableHelper` methods signatures remain the same.